internal/util/helpers_test.go

```diff
@@ -4,18 +4,758 @@
+}`,
+			expectedError: false,
+		},
+		{
+			name:          "Nil proto message",
+			input:         nil,
+			expectedJson:  `{}`,
+			expectedError: false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			jsonResult, err := util.GetJsonFromProto(tt.input)
+			if (err != nil) != tt.expectedError {
+				t.Errorf("GetJsonFromProto() error = %v, expectedError %v", err, tt.expectedError)
+				return
+			}
+			if jsonResult != tt.expectedJson {
+				t.Errorf("GetJsonFromProto() = %v, expected %v", jsonResult, tt.expectedJson)
+			}
+		})
+	}
+}
+
+// TestGetYamlFromProto tests the GetYamlFromProto function
+func TestGetYamlFromProto(t *testing.T) {
+	t.Parallel()
+	tests := []struct {
+		name          string
+		input         proto.Message
+		expectedYaml  string
+		expectedError bool
+	}{
+		{
+			name: "Valid proto message",
+			input: &pb.Repository{
+				Owner: "repoOwner",
+				Name:  "repoName",
+			},
+			expectedYaml: `name: repoName
+owner: repoOwner
+`,
+			expectedError: false,
+		},
+		{
+			name:  "Nil proto message",
+			input: nil,
+			expectedYaml: `{}
+`,
+			expectedError: false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			yamlResult, err := util.GetYamlFromProto(tt.input)
+			if (err != nil) != tt.expectedError {
+				t.Errorf("GetYamlFromProto() error = %v, expectedError %v", err, tt.expectedError)
+				return
+			}
+			if yamlResult != tt.expectedYaml {
+				t.Errorf("GetYamlFromProto() = %v, expected %v", yamlResult, tt.expectedYaml)
+			}
+		})
+	}
+}
+
+// TestOpenFileArg tests the OpenFileArg function
+func TestOpenFileArg(t *testing.T) {
+	t.Parallel()
+
+	testCases := []struct {
+		name         string
+		filePath     string
+		dashOpen     io.Reader
+		expectedDesc string
+		expectError  bool
+	}{
+		{
+			name:         "Dash as file path",
+			filePath:     "-",
+			dashOpen:     strings.NewReader("dash input"),
+			expectedDesc: "dash input",
+			expectError:  false,
+		},
+		{
+			name:         "Valid file path",
+			filePath:     "testfile.txt",
+			dashOpen:     nil,
+			expectedDesc: "test content",
+			expectError:  false,
+		},
+		{
+			name:         "Invalid file path",
+			filePath:     "nonexistent.txt",
+			dashOpen:     nil,
+			expectedDesc: "",
+			expectError:  true,
+		},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+		testDir := t.TempDir()
+		var tempFilePath string
+
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+			// Create a temporary file for testing
+			if tc.filePath == "testfile.txt" {
+				tempFilePath = filepath.Join(testDir, tc.filePath)
+				err := os.WriteFile(tempFilePath, []byte(tc.expectedDesc), 0600)
+				assert.NoError(t, err)
+				defer os.Remove(tempFilePath)
+			} else {
+				// When handling the dash (-) as a file path, it should read from the provided io.Reader (tc.dashOpen) instead of trying to open a file
+				tempFilePath = tc.filePath
+			}
```

Comment:
If you create `testfile.txt` outside the test case code, you can avoid needing as much conditional logic in the test.  Compare:

```golang
func TestOpenFileArg(t *testing.T) {
	t.Parallel()
	testDir := t.TempDir()
	// Create one test file for the file input case
	tempFilePath := filepath.Join(testDir, "testfile.txt")
	assert.NoError(os.WriteFile(tempFilePath, []byte("test content"), 0600))

	...
		filePath: tempFilePath
```

(vs the current 11 lines with `if`/`else`)

Reply:
thank you, I fixed it.